src/hiopbbpy/opt/boalgorithm.py

```diff
@@ -214,32 +261,43 @@ def optimize(self):
       y_new = np.array(y_new)
       y_train = np.vstack([y_train, y_new])
       
+      self.logger.scalars(f"Training set size is now {x_train.shape[0]}")
+      self.logger.iterations(f"Current best objective: {np.min(y_train):.4e} "
+                             f"(previous best: {prev_best:.4e})")
+      self.logger.scalars(f"Improvement: {prev_best - np.min(y_train):.4e}")
+
       # Save the new sample points and objective evaluations
       for j in range(1, self.batch_size+1):
         self.x_hist.append(x_train[-j].flatten())
         self.y_hist.append(y_train[-j].flatten())
+
       if self.batch_size == 1:
-        print(f"Sample point X:")
+        self.logger.debug(f"Sample point X:")
       else:
-        print(f"Sample points X:")
+        self.logger.debug(f"Sample points X:")
       for j in range(self.batch_size):
-        print(f"{x_train[-j-1]}")
+        self.logger.debug(f"  {x_train[-j-1]}")
+
       if self.batch_size == 1:
-        print(f"\nObservation Y:")
+        self.logger.debug(f"Observation Y:")
       else:
-        print(f"\nObservations Y:")
+        self.logger.debug(f"Observations Y:")
       for j in range(self.batch_size):
-        print(f"{y_new[-j-1]}")
+        self.logger.debug(f"  {y_new[-j-1]}")
 
     # Save the optimal results and all the training data
     self.idx_opt = np.argmin(self.y_hist)
     self.x_opt = self.x_hist[self.idx_opt]
     self.y_opt = self.y_hist[self.idx_opt]
     self.setTrainingData(x_train, y_train)
 
-    print(f"\n\nOptimal at BO iteration: {self.idx_opt+1} ")
-    print(f"Optimal point: {self.x_opt.flatten()}, Optimal value: {self.y_opt}")
-    print()
+    self.logger.critical("===================================")
+    self.logger.critical("Bayesian Optimization completed")
+    self.logger.critical(f"Total evaluations: {len(self.y_hist)}")
+    self.logger.critical(f"Optimal at BO iteration: {self.idx_opt//self.batch_size+1} ")
```

Comment:
When the batch_size is $1$ I think it makes more sense to say "Optimal at BO iteration: {self.idx_opt}/{self.bo_maxiter}". When the batch_size is greater than $1$ I think we need to be more careful as a batch of points are added per BO iteration and it's more complicated. E.g, if batch size is $5$ then the first $5$ points in `self.y_hist` are obtained in the first BO iteration and so self.idx_opt being $2$ doesn't mean that the optimal point was obtained in the $2$nd BO iteration.

Reply:
Thanks for clearing up my confusion. Thanks.